pyhf/tensor/mxnet_backend.py

```diff
@@ -105,11 +105,17 @@ def astensor(self, tensor_in, dtype='float'):
             MXNet NDArray: A multi-dimensional, fixed-size homogenous array.
         """
         dtypemap = {'float': 'float32', 'int': 'int32', 'bool': 'uint8'}
-        dtype = dtypemap[dtype]
         try:
-            tensor = nd.array(tensor_in, dtype=dtype)
-        except ValueError:
+            dtype = dtypemap[dtype]
+        except KeyError:
+            log.error('Invalid dtype: dtype must be float, int, or bool.')
+            raise
+
+        if isinstance(tensor_in, (int, float)):
```

Comment:
all these isinstance checks will slow down the code a little.. and `astensor` is something we call *a lot*

Reply:
Okay, I can revert them as we have gotten by without them with no problems.

Reply:
I guess this is something that we should discuss on your other comment as well, but it seemed that this would be a cleaner way to do it then my original method of

```python
try:
    tensor.shape[0]
except IndexError:
    tensor.function_to_expand_the_shape 
```

Do you think catching the `IndexError` is better, or do you think we shouldn't be trying at all?

Reply:
I think this is better at least... because then slowness is fixable by making sure we don't pass in scalars...

Reply:
Yes, EAFP is Pythonic and I'm definitely being all LBYL. I'll revert these to how I originally had them. :+1: